include/dlaf/eigensolver/reduction_to_band/impl.h

```diff
@@ -292,55 +293,53 @@ auto computePanelReflectors(MatrixLike& mat_a, const matrix::SubPanelView& panel
   }
 
   const std::size_t nthreads = getReductionToBandPanelNWorkers();
-  return ex::when_all(ex::just(std::make_shared<pika::barrier<>>(nthreads)),
-                      ex::just(std::vector<common::internal::vector<T>>{}),  // w (internally required)
-                      ex::just(common::internal::vector<T>{}),               // taus
-                      ex::when_all_vector(std::move(panel_tiles))) |
-         ex::transfer(di::getBackendScheduler<Backend::MC>(pika::execution::thread_priority::high)) |
-         ex::bulk(nthreads,
-                  [nthreads, nrefls, cols = panel_view.cols()](
-                      const std::size_t index, auto& barrier_ptr, auto& w, auto& taus, auto& tiles) {
-                    const auto barrier_busy_wait = getReductionToBandBarrierBusyWait();
-                    const std::size_t batch_size = util::ceilDiv(tiles.size(), nthreads);
-                    const std::size_t begin = index * batch_size;
-                    const std::size_t end = std::min(index * batch_size + batch_size, tiles.size());
-
-                    if (index == 0) {
-                      taus.reserve(nrefls);
-                      w.resize(nthreads);
-                    }
-
-                    for (SizeType j = 0; j < nrefls; ++j) {
-                      // STEP1: compute tau and reflector (single-thread)
-                      if (index == 0)
-                        taus.emplace_back(computeReflector(tiles, j));
-                      barrier_ptr->arrive_and_wait(barrier_busy_wait);
-
-                      // STEP2a: compute w (multi-threaded)
-                      const SizeType pt_cols = cols - (j + 1);
-                      if (pt_cols == 0)
-                        break;
-                      const bool has_head = (index == 0);
-
-                      w[index] = common::internal::vector<T>(pt_cols, 0);
-                      computeWTrailingPanel(has_head, tiles, w[index], j, pt_cols, begin, end);
-                      barrier_ptr->arrive_and_wait(barrier_busy_wait);
-
-                      // STEP2b: reduce w results (single-threaded)
-                      if (index == 0)
-                        dlaf::eigensolver::internal::reduceColumnVectors(w);
-                      barrier_ptr->arrive_and_wait(barrier_busy_wait);
-
-                      // STEP3: update trailing panel (multi-threaded)
-                      updateTrailingPanel(has_head, tiles, j, w[0], taus.back(), begin, end);
-                      barrier_ptr->arrive_and_wait(barrier_busy_wait);
-                    }
-                  }) |
-         ex::then([](auto barrier_ptr, auto w, auto taus, auto tiles) {
-           di::silenceUnusedWarningFor(barrier_ptr, w, tiles);
-           return taus;
-         }) |
-         ex::make_future();
+  auto s =
+      ex::when_all(ex::just(std::make_unique<pika::barrier<>>(nthreads),
+                            std::vector<common::internal::vector<T>>{}),  // w (internally required)
```

Comment:
`when_all(just(x), just(y), ...)` vs `when_all(just(x, y), ...)`:
Is just a matter of style, or is more efficient?

Reply:
Mostly style. I wouldn't object if you prefer the former stylistically. There's in theory a small difference in instantiations, function calls, and operation states created when using separate `just`s, but I wouldn't expect it to have a measurable impact at least at runtime. I have not measured it.

Reply:
No strong preference on my side.